fleet/fleet.go

```diff
@@ -103,6 +103,10 @@ type Fleet interface {
 	// GetStatusWithExpression fetches the current status of units based on a
 	// regular expression instead of a plain string.
 	GetStatusWithExpression(exp *regexp.Regexp) ([]UnitStatus, error)
+
+	// GetStatusWithMatcher returns a []UnitStatus, with an element for
+	// each unit where the given matcher returns true.
+	GetStatusWithMatcher(func(string) bool) ([]UnitStatus, error)
```

Comment:
No, we can actually delete the regexp version. But I didn't want to do immediately. We can clean this up.